prboom2/src/dsda/hud_components/big_ammo.c

```diff
@@ -42,9 +45,24 @@ static void dsda_DrawComponent(void) {
     return;
 
   ammo = player->ammo[ammo_type];
+  max_ammo = player->maxammo[weaponinfo[player->readyweapon].ammo];
+
+  if (ammo == max_ammo)
+    cm = CR_LIGHTBLUE;
+  else {
+    if (max_ammo)
```

Comment:
If you check above, you'll see that we don't reach this point in the code if there is no max ammo.